src/Maker/MakeScaffold.php

```diff
@@ -0,0 +1,288 @@
+    {
+        if ($input->getArgument('name')) {
+            return;
+        }
+
+        $scaffolds = $this->availableScaffolds();
+
+        while (true) {
+            $name = $io->choice('Available scaffolds', array_combine(
+                array_keys($scaffolds),
+                array_map(
+                    function (array $scaffold) {
+                        return $scaffold['description'];
+                    },
+                    $scaffolds
+                )
+            ));
+            $scaffold = $scaffolds[$name];
+
+            $io->title($name);
+            $io->text($scaffold['description']);
+            $io->newLine();
+
+            if ($scaffold['dependents'] ?? []) {
+                $io->text('This scaffold will also install the following scaffolds:');
+                $io->newLine();
+                $io->listing(array_map(
+                    function ($s) use ($scaffolds) {
+                        return sprintf('%s - %s', $s, $scaffolds[$s]['description']);
+                    },
+                    $scaffold['dependents'])
+                );
+            }
+
+            if ($scaffold['packages'] ?? []) {
+                $io->text('This scaffold will install the following composer packages:');
+                $io->newLine();
+                $io->listing(array_keys($scaffold['packages']));
+            }
+
+            if ($scaffold['js_packages'] ?? []) {
+                $io->text('This scaffold will install the following node packages:');
+                $io->newLine();
+                $io->listing(array_keys($scaffold['js_packages']));
+            }
+
+            if (!$io->confirm("Would your like to create the \"{$name}\" scaffold?")) {
+                $io->text('Going back to main menu...');
+
+                continue;
+            }
+
+            $input->setArgument('name', [$name]);
+
+            return;
+        }
+    }
+
+    private function generateScaffold(string $name, ConsoleStyle $io): void
+    {
+        if ($this->isScaffoldInstalled($name)) {
+            return;
+        }
+
+        if (!isset($this->availableScaffolds()[$name])) {
+            throw new \InvalidArgumentException("Scaffold \"{$name}\" does not exist for your version of Symfony.");
+        }
+
+        $scaffold = $this->availableScaffolds()[$name];
+
+        // install dependent scaffolds
+        foreach ($scaffold['dependents'] ?? [] as $dependent) {
+            $this->generateScaffold($dependent, $io);
+        }
+
+        $io->text("Installing <info>{$name}</info> Scaffold...");
+
+        // install required packages
+        foreach ($scaffold['packages'] ?? [] as $package => $env) {
+            if (!$this->isPackageInstalled($package)) {
+                $io->text("Installing Composer package: <comment>{$package}</comment>...");
+
+                $command = [$this->composerBin(), 'require', '--no-scripts', 'dev' === $env ? '--dev' : null, $package];
```

Comment:
I think we should also require packages using `--no-update` and do a `composer update package1 package2 ...` after the loop. Doing a single update will speed up the process for slower internet connections.